pkg/iterator/iter.go

```diff
@@ -122,6 +123,10 @@ func (a Args) Validate() error {
 		return fmt.Errorf("%w: invalid TokenExpirySecs", ErrInvalidArgs)
 	case a.MaxResults < 0:
 		return fmt.Errorf("%w: invalid MaxResults", ErrInvalidArgs)
+	case a.SANamespace == "" && a.SAName != "":
+		return fmt.Errorf("%w: ServiceAccountName provided but ServiceAccountNamespace missing", ErrInvalidArgs)
+	case a.SANamespace != "" && a.SAName == "":
+		return fmt.Errorf("%w: ServiceAccountNamespace provided but ServiceAccountName missing", ErrInvalidArgs)
```

Comment:
nit: The error message should refer to the fields - you're now calling them `SAName` and `SANamespace` and not `ServiceAccountName` and `ServiceAccountNamespace`